nixos/modules/tasks/filesystems/overlayfs.nix

```diff
@@ -73,10 +76,22 @@ let
         description = ''
           The path to the workdir.
 
+          If the filesystem is `neededForBoot`, this will be prefixed with `/sysroot`,
+          unless `prefixWithSysroot` is set to `false`.
+
           This MUST be set if you set `upperdir`.
         '';
       };
 
+      prefixWithSysroot = lib.mkOption {
+        type = lib.types.bool;
+        default = true;
+        description = ''
+          If enabled, mount point, `upperdir` and `workdir` will be prefixed with `/sysroot`.
```

Comment:
This isn't exactly true. The mount point will still have `/sysroot`, because that's added by `systemd-fstab-generator`, not by us.

Reply:
Indeed. Thanks for noticing this. Mounting things *in* the initrd is not in scope for this, and does not need to be, I think.